### ISSUE TITLE
Latest beta crash app

### ISSUE THREAD

**Issue by andreiiliuta** [devadvance/rootcloak#86]
Hi There,
I am using this app: https://play.google.com/store/apps/details?id=com.eTokenBCR

Before the latest beta the app was able to detect root.
With this beta I am not able to open the app after I add it into the Native section.
It just hangs and never starts.

Thanks for your help!

**Comment by pylerSM** [comment]
Yes, there is bug in Regex in native library. Do not use this feature yet

**Comment by andreiiliuta** [comment]
In the latest build it's not crashing anymore but the application still detects root unfortunately (even when using Native) :(
Can you turn this into an app request?
Thanks!

**Comment by pylerSM** [comment]
The best would be APK file or atleast native library *.so file

**Comment by andreiiliuta** [comment]
APK: https://www.dropbox.com/s/ivt5akhbutk94pk/base.apk?dl=0
.so file (I don't know if it's good): https://www.dropbox.com/s/psn39wse9dynd9z/libQRCronto.so?dl=0

After decompiling the app I think this is the file responsible for root: https://www.dropbox.com/s/q6gccfd47m2ni6h/sq.smali?dl=0

`.class public final Lsq;
.super Lts;
.source "SourceFile"


# instance fields
.field private a:Ljava/lang/String;


# direct methods
.method public constructor <init>()V
    .locals 2

    .prologue
    .line 17
    invoke-direct {p0}, Lts;-><init>()V

    .line 19
    const-string v0, "ignore"

    iput-object v0, p0, Lsq;->a:Ljava/lang/String;

    .line 21
    const-string v0, "RootDetection"

    const-string v1, "Your device is jailbroken or rooted. The application cannot be executed."

    invoke-virtual {p0, v0, v1}, Lsq;->c(Ljava/lang/String;Ljava/lang/String;)V
   

    .line 22
    return-void
.end method


# virtual methods
.method public final a()Ljava/lang/String;
    .locals 1

    .prologue
    .line 26
    iget-object v0, p0, Lsq;->a:Ljava/lang/String;

    return-object v0
.end method

.method public final a(Ljava/lang/String;)V
    .locals 0

    .prologue
    .line 31
    iput-object p1, p0, Lsq;->a:Ljava/lang/String;

    .line 32
    return-void
.end method
`
Thanks for your help!!

**Comment by pylerSM** [comment]
Take logcat during failed launch please

**Comment by andreiiliuta** [comment]
[etoken.txt](https://github.com/devadvance/rootcloak/files/384167/etoken.txt)
actually, it's still locking up when native is enabled. (the application freezes)
see attached logcat.

**Comment by pylerSM** [comment]
yes, sadly I know where the issue is. But we have fix.

**Comment by andreiiliuta** [comment]
I tried the latest beta and it's the same. After I install xposed, activate rootcloak and native library and try to run the eToken application, the eToken application is freezing.

**Comment by pylerSM** [comment]
Can you try https://github.com/devadvance/rootinspector/releases/download/v1.0/com.devadvance.rootinspector-v1.0.apk, remove native root detection for rootinspector and try to launch it?
